### PR TITLE
BSMP typehint & IO interface

### PR DESCRIPTION
**Pull request by carneirofc**

### CODE REVIEW

**Review by fernandohds564 — APPROVED** [review]

File: siriuspy/tests/bsmp/test_commands.py [hunk 1]
```diff
@@ -2,42 +2,50 @@
...
-        'binoperation_group',
-        'write_and_read_variable',
-        'create_group_of_variables',
-        'remove_all_groups_of_variables',
-        'request_curve_block',
-        'curve_block',
-        'recalculate_curve_checksum',
-        'execute_function',
-        'anomalous_response',
+        "entities",
```
Comment (fernandohds564) [inline siriuspy/tests/bsmp/test_commands.py:26]: @carneirofc, why double quotes? Single quotes are prettier, aren't they?

File: siriuspy/tests/bsmp/test_commands.py [hunk 2]
```diff
@@ -183,15 +193,15 @@ def test_read_variable_failure(self):
 
     def test_read_group_of_variables(self):
         """Test read_group_of_variables."""
-        ld_string = ['t', 'e', 's', 't', 'e']
+        ld_string = ["t", "e", "s", "t", "e"]
```
Comment (fernandohds564) [inline siriuspy/tests/bsmp/test_commands.py:196]: here too...

File: siriuspy/tests/bsmp/test_commands.py (hunk 1 above)
Comment (carneirofc) [reply siriuspy/tests/bsmp/test_commands.py:26]: I will roll back these changes. This ended up happening due to the linter/"code formatter" I use.
The double quotes is just an influence of other languages and also happens to be the default option on my linter of choice.

Comment (fernandohds564) [reply siriuspy/tests/bsmp/test_commands.py:26]: ok @carneirofc, thanks!

**Review by anacso17 — APPROVED** [review]

File: siriuspy/siriuspy/bsmp/entities.py [hunk 3]
```diff
@@ -1,34 +1,49 @@
...
+        self,
+        var_types: typing.Tuple[BSMPType],
+    ) -> int:
         size = 0
         try:
             for var_type in var_types:
                 size += var_type.size
         except TypeError:
-            return var_types.size
+            if isinstance(var_types, BSMPType):
```
Comment (carneirofc) [inline siriuspy/siriuspy/bsmp/entities.py:32]: @xresende  here I am including another check, so we don't assume the user is passing the correct type 
```python
BSMPType iterable | BSMPType
```

File: siriuspy/siriuspy/bsmp/entities.py [hunk 4]
```diff
@@ -39,43 +54,48 @@ def _conv_value_to_load(self, var_types, size, values):
...
-    def _conv_load_to_value(self, var_types, load):
+    def _conv_load_to_value(
+        self,
+        var_types: typing.Union[typing.Tuple[BSMPType], typing.List[BSMPType]],
+        load: typing.List[str]
+    ) -> typing.Union[str, float, int, typing.List[typing.Union[str, float, int]]]:
+        """Return a value or a list of values unpacked according to the BMSPType.fmt"""
         # NOTE: optimize this critical function!
-        load = list(map(ord, load))
+        _load = list(map(ord, load))
```
Comment (carneirofc) [inline siriuspy/siriuspy/bsmp/entities.py:64]: Rename the local variable `load` to `_load`

**Pull request merged**